packages/node-core/src/indexer/storeCache/storeCache.service.ts

```diff
@@ -83,9 +102,25 @@ export class StoreCacheService implements BeforeApplicationShutdown {
     return this.cachedModels[entity] as unknown as CachePoiModel;
   }
 
+  private async flushRelationalModelsInOrder(updatableModels: ICachedModelControl[], tx: Transaction): Promise<void> {
+    const relationalModels = updatableModels.filter((m) => m.hasAssociations);
+    // Extract all flushable records from each relational model and sort it
+    const sortedFlushableRecords = flatten(relationalModels.map((m) => m.flushableRecordsWithIndex())).sort(
+      (a, b) => a.operationIndex - b.operationIndex
+    );
+    // Handle record in operation order
+    for (const record of sortedFlushableRecords) {
+      const model = this.sequelize.model(record.entity);
+      if (record.action === IndexedOperationActionType.Set) {
+        await model.upsert(record.data, {transaction: tx});
+      } else if (record.action === IndexedOperationActionType.Remove) {
+        await model.destroy({where: {id: record.data.id} as any, transaction: tx});
+      }
+    }
```

Comment:
I think we can simplify this if we add a method to cache model. Something like `flushOperation(index: number)` then the model can handle the specific store operation and dealing with the model since we already have an instance of it there. 

Then we can have a really simple loop:

```ts
for (const i = 0; i < this._storeOperationIndex; i++) {
    // Flush operation can be a no-op if it doesn't hatve that index
    await Promise.all(this.models.map(m => m.flushOperation(i)));
}
```

I think it will also mean we can drop `IndexedOperationActionType` as the cacheModel knows what type of operation it is.

`flushOperation` should also remove the item from the cache as it has now been flushed, we don't seem to be doing that currently